unit-tests/syncer/sw.py

```diff
@@ -116,31 +185,50 @@ def expect( depth_frame = None, color_frame = None, nothing_else = False ):
     Looks at the syncer queue and gets the next frame from it if available, checking its contents
     against the expected frame numbers.
     """
-    global syncer
-    fs = syncer.poll_for_frames()  # NOTE: will never be None
-    if not fs:
+    global syncer, playback_status
+    f = syncer.poll_for_frame()
+    if playback_status is not None:
+        countdown = 50  # 5 seconds
+        while not f  and  playback_status != rs.playback_status.stopped:
+            countdown -= 1
+            if countdown == 0:
+                break
+            time.sleep( 0.1 )
+            f = syncer.poll_for_frame()
+    # NOTE: fs will never be None
+    if not f:
         test.check( depth_frame is None, "expected a depth frame" )
         test.check( color_frame is None, "expected a color frame" )
         return False
 
-    log.d( "Got", fs )
+    log.d( "Got", f )
+
+    fs = rs.composite_frame( f )
 
-    depth = fs.get_depth_frame()
+    if fs:
+        depth = fs.get_depth_frame()
+    else:
+        depth = rs.depth_frame( f )
     test.info( "actual depth", depth )
     test.check_equal( depth_frame is None, not depth )
     if depth_frame is not None and depth:
         test.check_equal( depth.get_frame_number(), depth_frame )
     
-    color = fs.get_color_frame()
+    if fs:
+        color = fs.get_color_frame()
+    elif not depth:
+        color = rs.video_frame( f )
+    else:
+        color = None
     test.info( "actual color", color )
     test.check_equal( color_frame is None, not color )
     if color_frame is not None and color:
         test.check_equal( color.get_frame_number(), color_frame )
```

Comment:
We compare a frame number to a frame?
Oh, I just realized that `color_frame` is actually `color_frame_number`
consider adding the arguments "_number" suffix

Reply:
Yeah it's a mess. If I change it I have to change lots of lines in alll 3 testcases. I'll do it separately.